lib/data-types.js

```diff
@@ -448,7 +480,7 @@ ENUM.prototype.key = ENUM.key = 'ENUM';
  * @property ARRAY
  */
 var ARRAY = function(type) {
-  var options = typeof type === "object" && !(type instanceof ABSTRACT) && type || {
+  var options = _.isPlainObject(type) && !(type instanceof ABSTRACT) && type || {
```

Comment:
We can probably remove ` !(type instanceof ABSTRACT)` with the change to `_.isPlainObject`